src/components/VideoPlayer.tsx

```diff
@@ -0,0 +1,61 @@
+import MuxPlayer, {MuxPlayerProps} from '@mux/mux-player-react'
+import {Card} from '@sanity/ui'
+import React, {PropsWithChildren, useMemo} from 'react'
+
+import {useClient} from '../hooks/useClient'
+import {MIN_ASPECT_RATIO} from '../util/constants'
+import {getVideoSrc} from '../util/getVideoSrc'
+import type {VideoAssetDocument} from '../util/types'
+import pluginPkg from './../../package.json'
+
+export default function VideoPlayer({
+  asset,
+  children,
+  ...props
+}: PropsWithChildren<
+  {asset: VideoAssetDocument; forceAspectRatio?: number} & Partial<Pick<MuxPlayerProps, 'autoPlay'>>
+>) {
+  const client = useClient()
+
+  const videoSrc = useMemo(() => asset.playbackId && getVideoSrc({client, asset}), [asset, client])
+
+  const signedToken = useMemo(() => {
+    try {
+      const url = new URL(videoSrc!)
+      return url.searchParams.get('token')
+    } catch {
+      return false
+    }
+  }, [videoSrc])
+
+  const [width, height] = (asset?.data?.aspect_ratio ?? '16:9').split(':').map(Number)
+  const targetAspectRatio =
+    props.forceAspectRatio || (Number.isNaN(width) ? 16 / 9 : width / height)
+  const aspectRatio = Math.max(MIN_ASPECT_RATIO, targetAspectRatio)
+  console.log({targetAspectRatio, MIN_ASPECT_RATIO, aspectRatio})
+
+  return (
+    <Card tone="transparent" style={{aspectRatio: aspectRatio, position: 'relative'}}>
+      <MuxPlayer
+        {...props}
+        playsInline
+        playbackId={`${asset.playbackId}${signedToken ? `?token=${signedToken}` : ''}`}
```

Comment:
Something goes wrong for signed assets (left hand side is this branch, right side is `main`):
![image](https://github.com/sanity-io/sanity-plugin-mux-input/assets/81981/3dd5de43-26fc-4168-80d5-5978c33e41a2)

Deploy link: https://sanity-plugin-mux-input-git-fork-hdoro-feat-better-navigation.sanity.build/production/desk/trailer;ecde4d39-52ac-4366-896f-c059dca5f17a

Error:
```
[mux-player 1.11.4] The specificed playback ID 1gj3tkQFA5CvTdxuWgu2jJS5Ms4Alzqk?token=**token** contains a token which must be provided via the playback-token attribute.
```

Seems to be because we have to use a new way of using JWT signed tokens: https://docs.mux.com/guides/player/advanced-usage#using-with-signed-urls

Reply:
Ahhh thank you for reproducing this, I forgot that signed URLs were a checkbox that I had access to and was hoping someone would test it for me 😅 

Issue fixed!